pkg/skaffold/deploy/kubectl/kubectl.go

```diff
@@ -92,15 +93,21 @@ func (k *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Art
 	// also, manually set the labels to ensure the runID is added
 	switch {
 	case len(k.hydratedManifests) > 0:
+		_, endTrace := instrumentation.StartTrace(ctx, "pkg/skaffold/deploy/kubectl", "CreateManifestList")
```

Comment:
So stylistic question: should we not have a trace for this entire `Deploy`, and then push these subtraces into the relevant functions?

Reply:
I have added this now.  My understanding was that the trace calls in applyResources:
```
func (r *SkaffoldRunner) applyResources(ctx context.Context, out io.Writer, artifacts, localImages []graph.Artifact) error {
	// Check that the cluster is reachable.
	// This gives a better error message when the cluster can't
	// be reached.
	if err := failIfClusterIsNotReachable(); err != nil {
		return fmt.Errorf("unable to connect to Kubernetes: %w", err)
	}

	ctx, endTrace := instrumentation.StartTrace(ctx, "pkg/skaffold/runner", "LoadImagesIntoCluster")
	if len(localImages) > 0 && r.runCtx.Cluster.LoadImages {
		err := r.loadImagesIntoCluster(ctx, out, localImages)
		if err != nil {
			endTrace()
			return err
		}
	}
	endTrace()

	deployOut, postDeployFn, err := deployutil.WithLogFile(time.Now().Format(deployutil.TimeFormat)+".log", out, r.runCtx.Muted())
	if err != nil {
		return err
	}

	event.DeployInProgress()
	ctx, endTrace = instrumentation.StartTrace(ctx, "pkg/skaffold/runner", "DeployInProgress")
	defer endTrace()
	namespaces, err := r.deployer.Deploy(ctx, deployOut, artifacts)
```
were perhaps sufficient (following Event V1 API methods) but it makes sense to add tracing here as well.   My initial approach was to put trace statements where event statements were for the larger commands (dev, build, deploy, etc.) but it might make sense to wrap funcitons directly vs putting trace calls w/ Event API calls

Reply:
After talking offline I will keep the current traces in the place as Event Calls to help with possible correlation of the two later